### PR TITLE
Add new blog post about Haskell Learning Group

### PR DESCRIPTION
**Pull request by HeeL**
New blog post "Haskell Learning Group: Our Progress and Updates"

### CODE REVIEW

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+* Basic DataTypes
+* Types
+* Typeclasses
+
+We learned important basics that helps a lot when you read and try to understand what certain code does. For instance understanding how to read type signature, so something like this starting to make sense:
+
+`(Num a, Num b) => a -> b -> b`
+
+In this case there are two constraints, one for each parameter. They must be of a typeclass Num. You might already have some idea what function might do, based on its type signature.
+We had a look on different examples of how type signatures limit a scope of what body of the function might be. Even more, some  type signatures of function tells us exactly what it does because its the only possible implementation of the function. The simplest example of that will be an identity function:
```
Comment (KamWis) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: `[... ]some  type signatures of function [...]` contains double space between `some` and `type` :)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+The only thing this function can possibly do is return parameter that was passed to it. More advanced example when there is only one possible implementation is function composition:
+
+```haskell
+f :: (b -> c) -> (a -> b) -> (a -> c)
+f x y = \z -> x(y(z))
+```
+
+It's not obvious from the first look at this type definition that there is no other possible implementation of a function based on its type. Compiler won't let us implement it in any other way.
```
Comment (KamWis) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: I don't know if I wouldn't stop at just giving `(Num a, Num b) => a -> b -> b` example w/o explaining this one and any further sample and complex type signature above.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+The only thing this function can possibly do is return parameter that was passed to it. More advanced example when there is only one possible implementation is function composition:
+
+```haskell
+f :: (b -> c) -> (a -> b) -> (a -> c)
+f x y = \z -> x(y(z))
+```
+
+It's not obvious from the first look at this type definition that there is no other possible implementation of a function based on its type. Compiler won't let us implement it in any other way. 
+
+## Katas and Exercises
```
Comment (KamWis) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: as much as I love this advice and your experience here - Maybe move it to new article? :) It would be very precious as a source of additional material for learning Haskell. Don't know, though, if this article is a good place for it.

But again - I would love to have it in separate article.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+I've started a [Haskell Exercises Repo][haskell-exercises-repo], which is basically nothing more than a collection of such problems and solutions to them. Most of them wouldn't make sense to make as a separate repository because of small size. So each directory is a separate exercise. It can be just one file or a Stack Project with tests and code that make them green. This repository has been created for learning purposes and accept contributions either as improvements to existing solutions or as new solutions. 
+
+[haskell-exercises-repo]: https://github.com/HeeL/haskell-exercises
+[exercism]: http://exercism.io
+[project-euler]: https://projecteuler.net
+
+## Haskell Meetup @ HolidayCheck
```
Comment (KamWis) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: This is awesome info! IMO it also would be handy to make this info as a poster and put it directly on the homepage or something. Because there is a chance that someone may not come across this one since this person isn't possibly interested in group meetings overall.

**Comment by KamWis** [comment]
Gr8 job man! Thank you for that :)

**Comment by KamWis** [comment]
ups. Miss clicked `Comment` with `Close and comment` :D

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 2 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: I understand your point, I was questioning this myself too. Last time I was criticized for explaining too much though

Comment (wolframkriesing) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: imho an article on this blog can be multi-purposes: recap (for/by Sergii) learning, teaching, summarizing and last but not least showing what cool stuff we do at HC.
And it has Sergii's touch, so I like the not pure focus on one thing. Which of course also should not prevent anyone from writing a pure "educational" article. #my2cents
Really cool effort Sergii!

**Comment by wolframkriesing** [comment]
makes me want to take part in the learning group :)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 1 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: I have no idea how you noticed that :) thx

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 3 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: idk, IMO this paragraph on its own is not worth to be a blogpost. The other option would be to remove it completely, but I see no reason why it doesn't belong here.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 4 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: yeah, I would also be interested in some analytics numbers from our blog. For instance how many people from Munich actually visiting our blog regularly.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 2 above)
Comment (KamWis) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: Effort is indeed cool :) It's just matter of to what we did agree with this blog post series - no one (I think) would stop @HeeL from adding new blogpost but from `how we learn` series :)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 3 above)
Comment (KamWis) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: It's just... like out of the context a bit - By all means - I like the idea, the advice you give here - I'm just afraid it won't stick to `group progress and updates`... Certainly it would be to short to call it `new article` - how about a little COOP so both of us could thing about making it bigger and more suitable as end-to-end guide for Haskell beginner?:)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 4 above)
Comment (KamWis) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: yeah, and I just taking about some additional info of such kind - as some print poster, or banner on the tech blog etc. - Let's spread the word out about this endeavour! :)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 1 above)
Comment (KamWis) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: hah! no problemo :) I just read this carefully and with interest! :)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+show_avatar : true
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2017-10-17-first-haskell-meeting/haskell-poster.jpg
+---
+
+This post is a quick update and the latest news about our Haskell learning group. Even though there were no blog posts to this topic recently, we are still having weekly meetings and mostly stick to the plan. That of course implies preparation time, spend between our sessions. There is no rush in our learning process and we take time to discuss interesting topics a bit longer, especially when all participants are involved. That being said, we might have a plan to go through one chapter per session but it also fine if it is less than that.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: On this topic

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+One thing we've noticed and that was to expect, is that the difficulty of a content is slowly increasing. We are spending more time reading chapters. Sometimes the same material has to be read multiple times in order to fully understand what is it about.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: As expected the difficulty is slowly increasing as we go along.

Your sentence structure is a bit confusing

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 5 above)
Comment (gilligan) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: implies -> includes? Even then it sounds a bit off. Maybe just drop it? Next sentence fits nicely

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+It helps a lot that the most of the concepts are familiar to us through other languages or libraries we are using. Let's say JavaScript doesn't support currying natively the way Haskell does it, but there are libraries which introduce these concepts. We are very often using [RamdaJS][ramdajs] in our projects, so such things as function composition, currying, partial application and so on were not something totally new to us. They were still had to be "translated" and demonstrated in a Haskell way but we certainly didn't need much time to get the idea.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: Not just “let’s say”, js does indeed not have  currying ;)

Comment (gilligan) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: So concepts such as ... were not completely foreign to us.

Comment (gilligan) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: They were still had to be?

Comment (gilligan) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: Are you talking about the syntax in Haskell which u had to get used to? Syntax and the type strictness? Would be good if u could be more specific

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+Similar situation was with typing system and type definitions. As many people from the group doing i.e. TypeScript or Scala, there were not that many surprises for us. Nevertheless it's pretty impressive what can be achieved with type constraints in Haskell.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: The whole paragraph reads a bit rough, will make suggestions next time i’m not on mobile..

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+[ramdajs]: http://ramdajs.com
+
+## Types
+
+We went through many topics in the book and since the last update we've covered chapter four, five and discussed a half of the chapter six which covers Typeclasses. The main topics were:
+* Basic DataTypes
+* Types
+* Typeclasses
+
+We learned important basics that helps a lot when you read and try to understand what certain code does. For instance understanding how to read type signature, so something like this starting to make sense:
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: Help, plural (basics)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 3 above)
Comment (gilligan) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: I like it.

Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: @KamWis I see your point, but following this principle, we might argue if Meetup section  belongs to this blogpost as well. I would prefer to keep it

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 4 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: We are often hosting all different kinds of MeetUps and workshops (i.e. Nix, JavaScript, Agile and so on). Maybe we create an "Events @ HC" section here and promote everything that happens at HolidayCheck?

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 5 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: `implies -> bedeuten` in this case. Idk, just wanted to point out that this group is a lot about doing things outside of our regular sessions.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 6 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: 😄 thx

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 7 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: > They were still had to be?

I guess this could be simplified to just "They still needed to be".
My intention was to express it like  "sie mussten übersetzt werden" (something like "they had to be translated" then)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 3 above)
Comment (KamWis) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: ok :)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 4 above)
Comment (KamWis) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: would be super nice - For now I am lost with what happens when. Good idea @HeeL

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 7 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: > Not just “let’s say”

I've meant like "for example". I guess I will change it to that

Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: > Are you talking about the syntax in Haskell which u had to get used to? Syntax and the type strictness? Would be good if u could be more specific

That was about things I've mentioned in the previous sentence, like function composition, currying, partial application and so on

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 9 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: In this case the word "help" is not about "basics" but about the fact that we've learned them, so I believe it should be singular.

**Comment by HeeL** [comment]
@KamWis  @gilligan 

thank you for your attention to my blog post. I've did rework after your suggestions. Would be nice if we can publish it soon, at least before 28th of November ;)

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 9 above)
Comment (gilligan) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: I am pretty sure the correct way to write this is

> We learned important basics that help us a lot [...]

The only thing that "help" can reference here are the important basics that we learned. Since you are talking about basics (plural) *they help us*. Hence "helps" -> "help". 

The only way around that would be a construct like: 

> Learning important basics such as A, B, C helps us to

because now what "helps" us is *the learning" and not "A, B, C".

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+show_avatar : true
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2017-10-17-first-haskell-meeting/haskell-poster.jpg
+---
+
+This post is a quick update and the latest news about our Haskell learning group. Even though there were no blog posts on this topic recently, we are still having weekly meetings and mostly stick to the plan. That of course implies preparation time, spend between our sessions. There is no rush in our learning process and we take time to discuss interesting topics a bit longer, especially when all participants are involved. That being said, we might have a plan to go through one chapter per session but it also fine if it is less than that.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: spend -> spent

You are talking about the time that is *being spent* on learning outside of the actual calls.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+As expected the difficulty is slowly increasing as we go along. We are spending more time reading chapters. Sometimes the same material has to be read multiple times in order to fully understand what is it about.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: what is it about -> what it is about.

Actually you maybe want to write "to fully understand it". Because i am pretty sure you understand the *context* (which is what it is about) but maybe not always all details.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+It helps a lot that the most of the concepts are familiar to us through other languages or libraries we are using. For instance JavaScript doesn't support currying natively the way Haskell does it, but there are libraries which introduce these concepts. We are very often using [RamdaJS][ramdajs] in our projects, so such things as function composition, currying, partial application and so on were not something totally new to us. They still had to be "translated" and demonstrated in a Haskell way but we certainly didn't need much time to get the idea.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: > doesn't support currying natively the way Haskell does it

JS doesn't have any kind of currying features. Not "natively" (not entirely sure the use of this word actually makes sense here?) or otherwise and not like Haskell either. Perhaps you could just rephrase it to something more like: 

"While Javascript functions are not curried, currying can can be simulated: libraries like Ramda or Lodash provide `curry` functions providing this functionality"

Also you are talking only about currying but then refer to currying as "these concepts" and only later introduce "function composition, partial application and so on"

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+Similar situation was with typing system and type definitions. As many people from the group doing i.e. TypeScript or Scala, there were not that many surprises for us. Nevertheless it's pretty impressive what can be achieved with type constraints in Haskell.
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: "Similar situation was with typing system" is a somewhat odd phrasing. 

Maybe try something like:

"The same applies to the the parts that we learned about the Haskell type system so far: ..."

I would be careful about not making too broad judgements. We have so far only talked only about the absolutely super tiny minimal first tidbits of the type system. The statement that "typing system and type definitions" are not surprising because people know Typescript and Scala might not be well received or sound silly. So it would probably help to restrict that statement a bit.

Comment (gilligan) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: typing system -> type system
"type constraints" is probably (?) not what you actually want to use? I think you want to refer to the type system in general?

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+```haskell
+id :: a -> a
+id x = x
+```
+
+The only thing this function can possibly do is return parameter that was passed to it. More advanced example when there is only one possible implementation is function composition:
+
+```haskell
+f :: (b -> c) -> (a -> b) -> (a -> c)
+f x y = \z -> x(y(z))
```
Comment (gilligan) [inline _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: you could write this as

```haskell
f g h = \j -> g $ h $ j
```
or even just by reusing the composition operator from Prelude:
```haskell
f g h = g . h
```

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 9 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: Ok, my guess was that I can implicitly refer to the "fact of learning" without mentioning it, but I guess you are right, I will write "help" because it refers to "basics"

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 11 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: "what it is about" 👍

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 12 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: "natively" -> "doesn't curry function arguments by default".

I see the problem here. I will rewrite it in another way

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 13 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: I've rewrote the sentence you've mentioned. 
As this blogpost meant to tell about our small steps in learning haskell, I think its fine to write about it from our perspective and base on what we know so far. I guess in next posts I can refer to this paragraph and say how wrong I was by thinking so :)

"type constraints" -> Constraints that are dictated by types, i.e. this argument must be an Integer.

File: _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md (hunk 14 above)
Comment (HeeL) [reply _posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md]: I had  these two alternative solutions in mind as I wrote this. I think its a good idea to mention them, but i.e. `g . h` is not something that was mentioned in the book.
